contracts/metatxToken/MetaTxToken.sol

```diff
@@ -0,0 +1,269 @@
+    }
+
+    function transfer(address dst, uint wad) public override returns (bool) {
+        return transferFrom(msgSender(), dst, wad);
+    }
+
+    function transferFrom(address src, address dst, uint wad)
+        public override virtual
+        returns (bool)
+    {
+        if (src != msgSender()) {
+            require(_approvals[src][msgSender()] >= wad, "ds-token-insufficient-approval");
+            _approvals[src][msgSender()] = sub(_approvals[src][msgSender()], wad);
+        }
+
+        require(_balances[src] >= wad, "ds-token-insufficient-balance");
+        _balances[src] = sub(_balances[src], wad);
+        _balances[dst] = add(_balances[dst], wad);
+
+        emit Transfer(src, dst, wad);
+
+        return true;
+    }
+
+    function approve(address guy, uint wad) public override returns (bool) {
+        _approvals[msgSender()][guy] = wad;
+
+        emit Approval(msgSender(), guy, wad);
+
+        return true;
+    }
+}
+
+
+
+
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+
+contract MetaTxToken is DSTokenBaseMeta(0), DSAuthMeta {
+  uint8 public decimals;
+  string public symbol;
+  string public name;
+
+  bool public locked;
+  bytes32 public DOMAIN_SEPARATOR;
+
+  mapping(address => uint256) metatransactionNonces;
+
+  function getMetatransactionNonce(address _user) override public view returns (uint256 nonce){
+    return metatransactionNonces[_user];
+  }
+
+  function incrementMetatransactionNonce(address _user) override internal {
+    metatransactionNonces[_user] = add(metatransactionNonces[_user], 1);
+  }
+
+  event Mint(address indexed guy, uint wad);
+  event Burn(address indexed guy, uint wad);
+
+  modifier unlocked {
+    if (locked) {
+      require(isAuthorized(msgSender(), msg.sig), "colony-token-unauthorised");
+    }
+    _;
+  }
+
+  constructor(string memory _name, string memory _symbol, uint8 _decimals) {
+    name = _name;
+    symbol = _symbol;
+    decimals = _decimals;
+    locked = true;
+
+    uint chainId;
+    assembly {
+        chainId := chainid()
+    }
+
+    DOMAIN_SEPARATOR = keccak256(
+        abi.encode(
+            keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
+            keccak256(bytes(name)),
+            keccak256(bytes("1")),
```

Comment:
Why?

Reply:
This is the version field in the [EIP-712](https://eips.ethereum.org/EIPS/eip-712) definition, which is used in the [PermittableTokenImplementation](https://blockscout.com/xdai/mainnet/address/0xf8D1677c8a0c961938bf2f9aDc3F3CFDA759A9d9/contracts) that tokens bridged on to xdai use. I'm mimicking that particular implementation so that the same Permit flow can be used in the front end for both types of tokens.